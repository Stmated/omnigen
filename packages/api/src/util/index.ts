export * from './Environment';

export type Arrayable<T> = T | Array<T>;
export type ReadonlyArrayable<T> = T | ReadonlyArray<T>;
