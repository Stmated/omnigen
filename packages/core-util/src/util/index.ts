
export * from './ProtocolHandler.js';
export * from './Sorters.js';
export * from './ToString.js';
export * from './Case.js';
export * from './Util.ts';
export * from './CombineTypeUtils.ts';

export function assertUnreachable(x: never): never {
  throw new Error(`Unreachable code was reached, with: ${getShallowPayloadString(x)}`);
}

export function assertDefined<T>(x: T): Exclude<T, undefined | null | void> {
  if (x === undefined || x === null) {
    throw new Error(`A reduced node became 'undefined' when it is required.`);
  }

  // @ts-ignore
  return x;
}

export function isDefined<T>(argument: T | undefined): argument is T {
  return argument !== undefined;
}

export function getShallowPayloadString<T>(origin: T, maxDepth = 1) {
  return getShallowPayloadStringInternal(origin, 0, maxDepth);
}

export function getShallowPayloadStringInternal<T>(origin: T, depth: number, maxDepth: number): string {

  return JSON.stringify(origin, (key, value) => {
    if (value && typeof value === 'object' && key) {
      if (depth < maxDepth) {
        return JSON.parse(getShallowPayloadStringInternal(value, depth + 1, maxDepth));
      }
      return '[...]';
    }
    return value;
  });
}
