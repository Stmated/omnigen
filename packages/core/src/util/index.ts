import {Simplify, Something} from './TsTypes.ts';

export * from './ProtocolHandler';
export * from './Sorters';
export * from './ToString';
export * from './Case';
export * from './Util';
export * from './CombineTypeUtils';
export * from './TsTypes';

export * from './assertions.ts';

export function assertUnreachable(x: never): never {
  throw new Error(`Unreachable code was reached, with: ${getShallowPayloadString(x)}`);
}

export function assertDefined<T>(x: T): Exclude<T, undefined | null | void> {
  if (x === undefined || x === null) {
    throw new Error(`A reduced node became 'undefined' when it is required.`);
  }

  // @ts-ignore
  return x;
}

export function isDefined<T>(argument: T | undefined | void): argument is T {
  return argument !== undefined;
}

export function getShallowPayloadString<T>(origin: T, maxDepth = 1) {
  return getShallowPayloadStringInternal(origin, 0, maxDepth);
}

export function getShallowPayloadStringInternal<T>(origin: T, depth: number, maxDepth: number): string {

  return JSON.stringify(origin, (key, value) => {
    if (value && typeof value === 'object' && key) {
      if (depth < maxDepth) {
        return JSON.parse(getShallowPayloadStringInternal(value, depth + 1, maxDepth));
      }
      return '[...]';
    }
    return value;
  });
}

function expectToBeInstanceOf<const T, const C extends new(...args: any[]) => T>(value: T, clss: C): asserts value is InstanceType<C> {
  if (!(value instanceof clss)) {
    throw new Error(`Expected value to be an instance of ${clss.name}`);
  }
}

export function expectToBeDefined<T>(val?: T): asserts val is Simplify<Something<T>> {
  if (!val) {
    throw new Error(`Expected value ${val} to be defined`);
  }
}

export function expectPropertyToBe<O, K extends keyof O, const V extends O[K]>(owner: O, key: K, compareVal: V): asserts owner is Extract<O, Record<K, V>> {
  if (owner[key] !== compareVal) {
    throw new Error(`${owner[key]} should have been ${compareVal}`);
  }
}

export interface ExpectTs {
  toBeDefined: typeof expectToBeDefined,
  toBeInstanceOf: typeof expectToBeInstanceOf,
  propertyToBe: typeof expectPropertyToBe,
}

export const expectTs: ExpectTs = {
  toBeDefined: expectToBeDefined,
  toBeInstanceOf: expectToBeInstanceOf,
  propertyToBe: expectPropertyToBe,
} as const;
