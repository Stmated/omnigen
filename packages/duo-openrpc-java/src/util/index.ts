
export * from './JavaTestUtils.ts';
export * from './OpenRpcTestUtils.ts';
