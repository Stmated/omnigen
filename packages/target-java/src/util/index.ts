export * from './JavaUtil';
export * from './FreeTextUtils';
