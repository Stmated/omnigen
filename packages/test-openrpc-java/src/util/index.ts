export * from './JavaTestUtils';
export * from './OpenRpcTestUtils';
export * from './ParsedJavaTestVisitor';
