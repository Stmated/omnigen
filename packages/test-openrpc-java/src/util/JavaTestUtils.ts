import * as JavaParser from 'java-parser';
import {ModelTransformOptions, PackageOptions, RenderedCompilationUnit} from '@omnigen/api';
import {JavaOptions, SerializationLibrary, ZodJavaOptions} from '@omnigen/target-java';
import {ParsedJavaTestVisitor} from '../util';
import {DEFAULT_SPECIFIC_TEST_TARGET_OPTIONS} from '@omnigen/utils-test';
import {Util, ZodCompilationUnitsContext} from '@omnigen/core';
import {PluginManager} from '@omnigen/plugin';
import {BaseContext, FileContext, TargetContext} from '@omnigen/core-plugin';
import {z, ZodObject} from 'zod';
import {LoggerFactory} from '@omnigen/core-log';
import {OpenRpcPlugins} from '@omnigen/parser-openrpc';

const logger = LoggerFactory.create(import.meta.url);

export const DEFAULT_SPECIFIC_TEST_JAVA_OPTIONS: Partial<JavaOptions> = {
  serializationLibrary: SerializationLibrary.POJO,
  beanValidation: false,
};

export const DEFAULT_TEST_JAVA_OPTIONS: JavaOptions = {
  ...ZodJavaOptions.parse({}),
  ...DEFAULT_SPECIFIC_TEST_TARGET_OPTIONS,
  ...DEFAULT_SPECIFIC_TEST_JAVA_OPTIONS,
};

export interface JavaTestUtilsOptions {
  // parserOptions?: Partial<ParserOptions>,
  modelTransformOptions?: Partial<ModelTransformOptions>,
  // targetOptions?: Partial<TargetOptions>,
  packageOptions?: Partial<PackageOptions>,
  javaOptions?: Partial<JavaOptions>,
  arguments?: Record<string, any>,
}

export class JavaTestUtils {

  constructor() {
    logger.debug(`Loaded OpenRPC: ${OpenRpcPlugins.OpenRpcPlugin}`);
  }

  public static async getFileContentsFromFile(fileName: string, options?: JavaTestUtilsOptions, source = 'openrpc'): Promise<Map<string, string>> {

    const filePath = Util.getPathFromRoot(`./packages/parser-${source}/examples/${fileName}`);
    const result = await JavaTestUtils.getResultFromFilePath(filePath, options || {}, ZodCompilationUnitsContext);

    return JavaTestUtils.cuToContentMap(result.compilationUnits);
  }

  public static async getResultFromFilePath<
    E extends ZodObject<any>,
    S extends ZodObject<any>
  >(
    filePath: string,
    options: JavaTestUtilsOptions,
    expected: E,
    stopAt?: S,
  ): Promise<z.output<E>> {

    const ctx: BaseContext & FileContext & TargetContext = {
      file: filePath,
      // target: 'java', // <-- implicit, because we have no other target plugin in this package
      defaults: {
        ...DEFAULT_SPECIFIC_TEST_TARGET_OPTIONS,
        ...DEFAULT_SPECIFIC_TEST_JAVA_OPTIONS,
      },
      arguments: {
        ...(options.modelTransformOptions),
        ...(options.packageOptions),
        ...(options.javaOptions),
        ...(options.arguments),
      },
    };

    const pm = new PluginManager({includeAuto: true});
    const result = await pm.execute({ctx: ctx, debug: true, stopAt: stopAt});
    const last = result.results[result.results.length - 1];

    return expected.parse(last.ctx) as z.output<E>;
  }

  public static getParsedContent(fileContents: Map<string, string>, fileName: string): ParsedJavaTestVisitor {

    const content = fileContents.get(fileName) || '';
    let cst: JavaParser.CstNode;
    try {
      cst = JavaParser.parse(content);
      if (!cst) {
        throw new Error(`The JavaParser must return something`);
      }

    } catch (ex) {
      throw new Error(`Could not parse '${fileName}'': ${ex}:\n${content}`, {cause: ex instanceof Error ? ex : undefined});
    }

    const visitor = new ParsedJavaTestVisitor();
    visitor.visit(cst);

    return visitor;
  }

  public static cuToContentMap(compilationUnits: RenderedCompilationUnit[]) {
    const fileContents = new Map<string, string>();

    for (const render of compilationUnits) {
      fileContents.set(render.fileName, render.content);
    }

    return fileContents;
  }
}
