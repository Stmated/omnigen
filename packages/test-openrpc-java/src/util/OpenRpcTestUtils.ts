import fs from 'fs/promises';
import {JavaTestUtils, JavaTestUtilsOptions} from './JavaTestUtils.ts';
import {OmniModelParserResult, PackageOptions, TargetOptions} from '@omnigen/api';
import {JavaPlugins, JavaOptions} from '@omnigen/target-java';
import {ZodModelContext, ZodPackageOptionsContext, ZodTargetOptionsContext} from '@omnigen/core-plugin';
import {Util, ZodCompilationUnitsContext} from '@omnigen/core';

export type KnownSchemaNames = 'openrpc';

/**
 * TODO: REMOVE THIS WHOLE THING! OR AT LEAST SERIOUSLY MINIMIZE IT! Use `Omnigen` base class instead!
 *        `Omnigen` needs to be made more dynamic/non-hardcoded first, then this class can just help with some basic options/args!
 */
export class OpenRpcTestUtils {

  static getKnownSchemaNames(): KnownSchemaNames[] {
    return ['openrpc'];
  }

  static async listExampleFileNames(type: KnownSchemaNames): Promise<string[]> {
    const dirPath = Util.getPathFromRoot(`./packages/parser-${type}/examples/`);
    return fs.readdir(dirPath, {withFileTypes: true})
      .then(paths => {
        return paths.filter(it => it.isFile()).map(it => it.name);
      });
  }

  /**
   * TODO: Need a way to state what the "final" plugin should be
   *        since we should NOT render things here, we only care about the generated model
   */
  static async readExample(
    type: KnownSchemaNames,
    fileName: string,
    options: JavaTestUtilsOptions,
  ): Promise<OmniModelParserResult<JavaOptions & PackageOptions & TargetOptions>> {

    const result = await JavaTestUtils.getResultFromFilePath(
      Util.getPathFromRoot(`./packages/parser-${type}/examples/${fileName}`),
      options,
      ZodModelContext
        .merge(JavaPlugins.ZodJavaOptionsContext)
        .merge(ZodPackageOptionsContext)
        .merge(ZodTargetOptionsContext),
      ZodCompilationUnitsContext,
    );

    return {
      model: result.model,
      options: {
        ...result.javaOptions,
        ...result.packageOptions,
        ...result.targetOptions,
      },
    };
  }
}
