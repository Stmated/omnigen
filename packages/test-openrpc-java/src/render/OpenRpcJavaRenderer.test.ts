import * as fs from 'fs';
import * as path from 'path';
import * as JavaParser from 'java-parser';
import {JavaTestUtils, OpenRpcTestUtils, ParsedJavaTestVisitor} from '../util';
import {describe, test, vi} from 'vitest';
import {OmniTypeKind} from '@omnigen/api';
import {Util, ZodCompilationUnitsContext} from '@omnigen/core';
import {LoggerFactory} from '@omnigen/core-log';
import {SerializationLibrary} from '@omnigen/target-java';
import {SerializationPropertyNameMode} from '@omnigen/target-code';

const logger = LoggerFactory.create(import.meta.url);

describe('Java Rendering', () => {

  test.concurrent('renderAll', async ctx => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    // TODO: Do this both with and without compressing types -- create different versions and output them in a structure!
    // TODO: Would it be crazy or cool to take ALL the types of all the models and create one HUGE output with common types?

    const errors: Error[] = [];

    for (const schemaName of OpenRpcTestUtils.getKnownSchemaNames()) {

      let fileNames: string[];
      try {
        fileNames = await OpenRpcTestUtils.listExampleFileNames(schemaName);
      } catch (ex) {
        throw new Error(`Could not list filenames inside ${schemaName}: ${ex}`, {cause: ex instanceof Error ? ex : undefined});
      }

      for (const fileName of fileNames) {

        try {

          logger.info(`Will render ${fileName}`);
          const result = await JavaTestUtils.getResultFromFilePath(
            Util.getPathFromRoot(`./packages/parser-openrpc/examples/${fileName}`), {}, ZodCompilationUnitsContext,
          );

          let baseDir: string;

          const targetTestDir = path.resolve('./.target_test');

          try {
            baseDir = path.resolve(`./.target_test/${schemaName}/${path.basename(fileName, path.extname(fileName))}`);
          } catch (ex) {
            errors.push(new Error(`Could not resolve path of ${fileName}`, {cause: ex}));
            continue;
          }

          for (const cu of result.compilationUnits) {

            if (cu.fileName.indexOf('#') !== -1) {
              throw new Error(`'#' not allowed in CU '${cu.fileName}'`);
            }

            const outDir = `${baseDir}/${cu.directories.join('/')}`;
            if (!fs.existsSync(outDir)) {
              fs.mkdirSync(outDir, {recursive: true});
            }

            const outPath = `${outDir}/${cu.fileName}`;

            try {
              fs.writeFileSync(outPath, cu.content);
            } catch (ex) {
              errors.push(new Error(`Could not write '${outPath}' of '${fileName}': ${ex}`, {cause: ex}));
              continue;
            }

            let cst: JavaParser.CstNode;
            try {
              cst = JavaParser.parse(cu.content);
              ctx.expect(cst).toBeDefined();
            } catch (ex) {
              errors.push(new Error(`Could not parse '${schemaName}' '${fileName}' in '${outPath}': ${ex}\n\n${cu.content}\n\n`, {cause: ex}));
              continue;
            }

            // Visit the syntax tree, but do nothing with the result.
            // It is only to see if any crashes can be found or not.
            const visitor = new ParsedJavaTestVisitor();
            visitor.visit(cst);
          }

          // Everything went fine. So we delete the test output directory. If things went badly it will stay around.
          fs.rmSync(targetTestDir, {recursive: true, force: true});

        } catch (ex) {
          errors.push(LoggerFactory.formatError(ex, `File ${fileName}`));
        }
      }
    }

    if (errors.length > 0) {
      expect.fail(`Rendering errors:\n* ${errors.map(it => `${it.message}:\n${it.stack}`).join('\n\n* ')}`);
    }
  }, {
    timeout: 30_000,
  });

  test.concurrent('Test multiple inheritance (interfaces)', async ctx => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const fileContents = await JavaTestUtils.getFileContentsFromFile('multiple-inheritance.json', {
      javaOptions: {
        serializationLibrary: SerializationLibrary.JACKSON,
        serializationPropertyNameMode: SerializationPropertyNameMode.IF_REQUIRED,
      },
    });

    ctx.expect([...fileContents.keys()].sort()).toMatchSnapshot();
    for (const [fileName, fileContent] of fileContents) {
      ctx.expect(fileContent).toMatchFileSnapshot(`./__snapshots__/${ctx.task.suite?.name}/${ctx.task.name}/${fileName}`);
    }
  });

  test.concurrent('Type compressions', async ctx => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    // Check that the property 'common' from A and B are moved into Abs.

    const fileContents = await JavaTestUtils.getFileContentsFromFile('compressable-types.json', {
      modelTransformOptions: {generifyTypes: false},
      javaOptions: {
        preferNumberType: OmniTypeKind.DOUBLE,
        serializationLibrary: SerializationLibrary.JACKSON,
        serializationPropertyNameMode: SerializationPropertyNameMode.IF_REQUIRED,
        beanValidation: false,
      },
    });

    ctx.expect([...fileContents.keys()].sort()).toMatchSnapshot();
    for (const [fileName, fileContent] of fileContents) {
      ctx.expect(fileContent).toMatchFileSnapshot(`./__snapshots__/${ctx.task.suite?.name}/${ctx.task.name}/${fileName}`);
    }
  });

  test.concurrent('Enum', async ctx => {

    const fileContents = await JavaTestUtils.getFileContentsFromFile('enum.json', {
      javaOptions: {
        includeGenerated: false,
        serializationLibrary: SerializationLibrary.JACKSON,
        serializationPropertyNameMode: SerializationPropertyNameMode.IF_REQUIRED,
        singleFile: true,
        singleFileName: ctx.task.name,
        // debug: true,
      },
    });

    ctx.expect([...fileContents.keys()].sort()).toMatchSnapshot();
    for (const [fileName, fileContent] of fileContents) {
      ctx.expect(fileContent).toMatchFileSnapshot(`./__snapshots__/${ctx.task.suite?.name}/${fileName}`);
    }
  });

  test.concurrent('AdditionalProperties', async ctx => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const fileContents = await JavaTestUtils.getFileContentsFromFile('additional-properties.json', {
      javaOptions: {
        serializationLibrary: SerializationLibrary.JACKSON,
        serializationPropertyNameMode: SerializationPropertyNameMode.IF_REQUIRED,
        singleFile: true,
        singleFileName: ctx.task.name,
        additionalPropertiesInterfaceAfterDuplicateCount: 1,
      },
    });

    ctx.expect([...fileContents.keys()].sort()).toMatchSnapshot();
    for (const [fileName, fileContent] of fileContents) {
      ctx.expect(fileContent).toMatchFileSnapshot(`./__snapshots__/${ctx.task.suite?.name}/${fileName}`);
    }
  });

  test.concurrent('description-inheritance', async ctx => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const fileContents = await JavaTestUtils.getFileContentsFromFile('description-inheritance.json', {
      javaOptions: {
        singleFile: true,
        singleFileName: 'description-inheritance',
      },
    });

    ctx.expect([...fileContents.keys()].sort()).toMatchSnapshot();
    for (const [fileName, fileContent] of fileContents) {
      ctx.expect(fileContent).toMatchFileSnapshot(`./__snapshots__/${ctx.task.suite?.name}/${fileName}`);
    }
  });

  test.concurrent('PetStore-Expanded-ClassNames', async ctx => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const fileContents = await JavaTestUtils.getFileContentsFromFile('petstore-expanded.json');
    const filenames = [...fileContents.keys()];

    ctx.expect(filenames).toContain('Pet.java');
    ctx.expect(filenames).toContain('DeletePetByIdResponse.java');
    ctx.expect(filenames).not.toContain('Pet1.java');
  });

  test.concurrent('primitive-generics', async ctx => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const fileContents = await JavaTestUtils.getFileContentsFromFile('primitive-generics.json', {
      javaOptions: {preferNumberType: OmniTypeKind.DOUBLE, serializationLibrary: SerializationLibrary.POJO},
    });

    ctx.expect([...fileContents.keys()].sort()).toMatchSnapshot();
    ctx.expect(fileContents.get('GiveNumberGetCharRequestParams.java')).toMatchSnapshot();

    // TODO: Add more exact checks for all generic types, making sure they have the correct amount of generics and whatever
  });

  test.concurrent('generic_params', async ctx => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const result = await JavaTestUtils.getResultFromFilePath(Util.getPathFromRoot(`./packages/test-openrpc-java/examples/generic_params.json`), {
      javaOptions: {
        defaultAdditionalProperties: false,
        serializationLibrary: SerializationLibrary.JACKSON,
        serializationPropertyNameMode: SerializationPropertyNameMode.ALWAYS,
        compressSoloReferencedTypes: true,
        compressUnreferencedSubTypes: true,
        additionalPropertiesInterfaceAfterDuplicateCount: 100,
      },
    }, ZodCompilationUnitsContext);

    const fileContents = await JavaTestUtils.cuToContentMap(result.compilationUnits);

    ctx.expect([...fileContents.keys()].sort()).toMatchSnapshot();
    for (const [fileName, fileContent] of fileContents) {
      ctx.expect(fileContent).toMatchFileSnapshot(`./__snapshots__/${ctx.task.suite?.name}/${ctx.task.name}/${fileName}`);
    }
  });

  test.concurrent('method-in-response', async ctx => {

    vi.useFakeTimers({now: new Date('2000-01-02T03:04:05.000Z')});

    const fileContents = await JavaTestUtils.getFileContentsFromFile('method-in-response.json', {
      javaOptions: {
        singleFile: true,
        singleFileName: ctx.task.name,
      },
    });

    ctx.expect([...fileContents.keys()].sort()).toMatchSnapshot();
    for (const [fileName, fileContent] of fileContents) {
      ctx.expect(fileContent).toMatchFileSnapshot(`./__snapshots__/${ctx.task.suite?.name}/${ctx.task.name}/${fileName}`);
    }
  });

  // TODO: Create test case that actually needs the expandedGenericSourceIdentifier inside GenericsModelTransformer -- it is a good feature, just needs some fixes
});
