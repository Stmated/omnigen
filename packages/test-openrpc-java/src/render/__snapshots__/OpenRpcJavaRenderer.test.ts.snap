// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Java Rendering > AdditionalProperties 1`] = `
[
  "AdditionalProperties.java",
]
`;

exports[`Java Rendering > Enum 1`] = `
[
  "Enum.java",
]
`;

exports[`Java Rendering > Test multiple inheritance (interfaces) 1`] = `
[
  "A.java",
  "Abs.java",
  "B.java",
  "C.java",
  "ErrorUnknown.java",
  "ErrorUnknownError.java",
  "GiveInGetOut2Request.java",
  "GiveInGetOut2RequestParams.java",
  "GiveInGetOut2Response.java",
  "GiveInGetOutRequest.java",
  "GiveInGetOutRequestParams.java",
  "GiveInGetOutResponse.java",
  "IB.java",
  "IC.java",
  "In.java",
  "JsonRpcError.java",
  "JsonRpcErrorResponse.java",
  "JsonRpcRequest.java",
  "JsonRpcRequestParams.java",
  "JsonRpcResponse.java",
  "Out.java",
  "Out2.java",
  "UnionOfAB.java",
]
`;

exports[`Java Rendering > Type compressions 1`] = `
[
  "A.java",
  "Abs.java",
  "B.java",
  "ErrorUnknown.java",
  "ErrorUnknownError.java",
  "GiveIn1GetOut1Request.java",
  "GiveIn1GetOut1RequestParams.java",
  "GiveIn1GetOut1Response.java",
  "GiveIn2GetOut2Request.java",
  "GiveIn2GetOut2RequestParams.java",
  "GiveIn2GetOut2Response.java",
  "In1.java",
  "In2.java",
  "JsonRpcError.java",
  "JsonRpcErrorResponse.java",
  "JsonRpcRequest.java",
  "JsonRpcRequestParams.java",
  "JsonRpcResponse.java",
]
`;

exports[`Java Rendering > description-inheritance 1`] = `
[
  "description-inheritance.java",
]
`;

exports[`Java Rendering > generic_params 1`] = `
[
  "AbstractRequestData.java",
  "DepositRequest.java",
  "DepositResponse.java",
  "ErrorUnknown.java",
  "ErrorUnknownError.java",
  "JsonRpcError.java",
  "JsonRpcErrorResponse.java",
  "JsonRpcRequest.java",
  "JsonRpcRequestParams.java",
  "JsonRpcResponse.java",
  "RefundRequest.java",
  "RefundResponse.java",
]
`;

exports[`Java Rendering > primitive-generics 1`] = `
[
  "ErrorUnknown.java",
  "ErrorUnknownError.java",
  "GiveIntGetDoubleRequest.java",
  "GiveIntGetDoubleRequestParams.java",
  "GiveIntGetDoubleResponse.java",
  "GiveNumberGetCharRequest.java",
  "GiveNumberGetCharRequestParams.java",
  "GiveNumberGetCharResponse.java",
  "GiveStringGetStringRequest.java",
  "GiveStringGetStringRequestParams.java",
  "GiveStringGetStringResponse.java",
  "JsonRpcError.java",
  "JsonRpcErrorResponse.java",
  "JsonRpcRequest.java",
  "JsonRpcRequestParams.java",
  "JsonRpcResponse.java",
]
`;

exports[`Java Rendering > primitive-generics 2`] = `
"package generated.omnigen;

import jakarta.annotation.Generated;

@Generated(value = "omnigen", date = "2000-01-02T03:04:05.000Z")
public class GiveNumberGetCharRequestParams extends JsonRpcRequestParams<Double> {
  public GiveNumberGetCharRequestParams(double param) {
    super(param);
  }
}
"
`;
